relayer/src/chain/cosmos.rs

```diff
@@ -200,30 +200,46 @@ impl CosmosSdkChain {
                 ));
             }
 
+            Ok(())
+        }
+
+        if let Err(e) = self.block_on(do_health_checkup(self)) {
+            warn!("Health checkup for chain '{}' failed", self.id());
+            warn!("Reason: {}", e);
+            warn!("Some Hermes features may not work in this mode!");
+        }
+    }
+
+    pub fn validate_params(&self) {
+        fn do_validate_params(chain: &CosmosSdkChain) -> Result<(), Error> {
             // Check on the configured max_tx_size against genesis block max_bytes parameter
-            let a = chain.rpc_client.genesis().await.map_err(|e| {
+            let genesis = chain.block_on(chain.rpc_client.genesis()).map_err(|e| {
                 Error::health_check_json_rpc(
-                    chain_id.clone(),
-                    rpc_address.clone(),
+                    chain.id().clone(),
+                    chain.config.rpc_addr.to_string(),
                     "/genesis".to_string(),
                     e,
                 )
             })?;
-            let max_allowed = a.consensus_params.block.max_bytes as f64 * 0.9;
-            if max_tx_size >= max_allowed {
-                return Err(Error::health_check_tx_size_out_of_bounds(
-                    chain_id.clone(),
-                    max_tx_size,
-                    a.consensus_params.block.max_bytes,
+
+            let genesis_max_bound = genesis.consensus_params.block.max_bytes;
+            let max_allowed = mul_ceil(genesis_max_bound, 0.9) as usize;
+
+            if chain.max_tx_size() > max_allowed {
+                return Err(Error::tx_size_out_of_bounds(
+                    chain.id().clone(),
+                    chain.max_tx_size(),
+                    genesis_max_bound,
                 ));
             }
 
             Ok(())
         }
 
-        if let Err(e) = self.block_on(do_health_checkup(self, self.max_tx_size() as f64)) {
-            warn!("{}", e);
-            warn!("some Hermes features may not work in this mode!");
+        if let Err(e) = do_validate_params(self) {
+            warn!("Hermes might be misconfigured for chain '{}'", self.id());
+            warn!("Reason: {}", e);
+            warn!("Some Hermes features may not work in this mode!");
```

Comment:
I like the separation of `chain.health_checkup` from `chain.validate_params`.

Note that `do_validate_params` may throw one of two errors:

- `health_check_json_rpc`
- `tx_size_out_of_bounds`

The warning "Hermes might be misconfigured for chain" ... makes sense for the second message, but not for the first one. This is an extremely minor issue, however, so it should not block this PR!

Here is how the log looks like


```
Jul 29 14:23:24.470  WARN Health checkup for chain 'ibc-0' failed
Jul 29 14:23:24.470  WARN Reason: Hermes health check failed for endpoint /health on the Json RPC interface of chain ibc-0:http://127.0.0.1:26657/
Jul 29 14:23:24.470  WARN Some Hermes features may not work in this mode!
Jul 29 14:23:24.471  WARN Hermes might be misconfigured for chain 'ibc-0'
Jul 29 14:23:24.471  WARN Reason: Hermes health check failed for endpoint /genesis on the Json RPC interface of chain ibc-0:http://127.0.0.1:26657/
```

Reply:
Right, let me fix this!

Reply:
Done

Reply:
Also improved the layout of the errors a bit:

```
Jul 29 14:59:29.177  WARN Health checkup for chain 'ibc-0' failed
Jul 29 14:59:29.177  WARN     Reason: health check failed for endpoint /health on the JSON-RPC interface of chain ibc-0:http://127.0.0.1:26657/
Jul 29 14:59:29.177  WARN     Some Hermes features may not work in this mode!
Jul 29 14:59:29.178  WARN Hermes might be misconfigured for chain 'ibc-0'
Jul 29 14:59:29.179  WARN     Reason: semantic config validation failed for endpoint /genesis on the JSON-RPC interface of chain ibc-0:http://127.0.0.1:26657/
Jul 29 14:59:29.179  WARN     Some Hermes features may not work in this mode!
```